includes/Core/Site_Health/Tags_Placement.php

```diff
@@ -0,0 +1,257 @@
+
+	use Method_Proxy_Trait;
+
+	/**
+	 * Modules instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Modules
+	 */
+	private $modules;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Modules $modules Modules instance.
+	 */
+	public function __construct( Modules $modules ) {
+		$this->modules = $modules;
+	}
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		add_filter(
+			'googlesitekit_rest_routes',
+			function ( $rest_routes ) {
+				$health_check_routes = $this->get_rest_routes();
+
+				return array_merge( $rest_routes, $health_check_routes );
+			}
+		);
+
+		add_filter(
+			'site_status_tests',
+			function ( $tests ) {
+				global $wp_version;
+
+				if ( version_compare( $wp_version, '5.6', '<' ) ) {
+					$tests['direct']['tag_placement'] = array(
+						'label' => __( 'Tags Placement', 'google-site-kit' ),
+						'test'  => $this->get_method_proxy( 'tags_placement_test' ),
+					);
+
+					return $tests;
+				}
+
+				$tests['async']['tag_placement'] = array(
+					'label'             => __( 'Tags Placement', 'google-site-kit' ),
+					'test'              => rest_url( 'google-site-kit/v1/core/site/data/tags-placement-test' ),
+					'has_rest'          => true,
+					'async_direct_test' => $this->get_method_proxy( 'tags_placement_test' ),
+				);
+
+				return $tests;
+			}
+		);
+	}
+
+	/**
+	 * Gets all REST routes.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return REST_Route[] List of REST_Route objects.
+	 */
+	private function get_rest_routes() {
+		return array(
+			new REST_Route(
+				'core/site/data/tags-placement-test',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => $this->get_method_proxy( 'tags_placement_test' ),
+						'permission_callback' => function () {
+							return current_user_can( Permissions::SETUP );
+						},
+					),
+				)
+			),
+		);
+	}
+
+	/**
+	 * Checks if the modules tags are placed on the website.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Site health status results.
+	 */
+	protected function tags_placement_test() {
+		global $wp_version;
+
+		$result = array(
+			'label'   => __( 'Tags Placement', 'google-site-kit' ),
+			'status'  => 'recommended',
+			'badge'   => array(
+				'label' => __( 'Site Kit', 'google-site-kit' ),
+				'color' => 'blue',
+			),
+			'actions' => '',
+			'test'    => 'tag_placement',
+		);
+
+		if ( version_compare( $wp_version, '5.6', '<' ) ) {
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'This feature requires WordPress version 5.6 or higher', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
+		$active_modules = $this->get_active_modules();
+		if ( empty( $active_modules ) ) {
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'Tag status not available: AdSense, Tag Manager, and Analytics modules are not connected.', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
```

Comment:
Note there are other reasons why tags may not be output which would be normal and expected. i.e. https://github.com/google/site-kit-wp/blob/d952edb61a7604dede836bc3df8c8b8eddc2afbd/includes/Core/Tags/Guards/Tag_Environment_Type_Guard.php

or `useSnippet` is turned off.